cairo/ethereum/utils/numeric.cairo

```diff
@@ -336,3 +383,179 @@ func Bytes32_from_be_bytes{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(bytes:
+        memset_zero(bytes_ptr, d3_bytes_len - d3_actual_len);
+        memcpy(bytes_ptr + d3_bytes_len - d3_actual_len, d3_bytes, d3_actual_len);
+
+        // Copy limb2 with padding to exactly 12 bytes
+        memset_zero(bytes_ptr + d3_bytes_len, 12 - d2_actual_len);
+        memcpy(bytes_ptr + d3_bytes_len + 12 - d2_actual_len, d2_bytes, d2_actual_len);
+
+        // Copy limb1 with padding to exactly 12 bytes
+        memset_zero(bytes_ptr + d3_bytes_len + 12, 12 - d1_actual_len);
+        memcpy(bytes_ptr + d3_bytes_len + 12 + 12 - d1_actual_len, d1_bytes, d1_actual_len);
+
+        // Copy limb0 with padding to exactly 12 bytes
+        memset_zero(bytes_ptr + d3_bytes_len + 24, 12 - d0_actual_len);
+        memcpy(bytes_ptr + d3_bytes_len + 24 + 12 - d0_actual_len, d0_bytes, d0_actual_len);
+
+        tempvar range_check_ptr = range_check_ptr;
+    } else {
+        let is_length_gt_24 = is_le(24, length);
+        if (is_length_gt_24 != 0) {
+            // Case: mod_len > 12 * 2
+            // Copy limb2 with padding
+            let d2_bytes_len = length - 24;
+            memset_zero(bytes_ptr, d2_bytes_len - d2_actual_len);
+            memcpy(bytes_ptr + d2_bytes_len - d2_actual_len, d2_bytes, d2_actual_len);
+
+            // Copy limb1 with padding to exactly 12 bytes
+            memset_zero(bytes_ptr + d2_bytes_len, 12 - d1_actual_len);
+            memcpy(bytes_ptr + d2_bytes_len + 12 - d1_actual_len, d1_bytes, d1_actual_len);
+
+            // Copy limb0 with padding to exactly 12 bytes
+            memset_zero(bytes_ptr + d2_bytes_len + 12, 12 - d0_actual_len);
+            memcpy(bytes_ptr + d2_bytes_len + 12 + 12 - d0_actual_len, d0_bytes, d0_actual_len);
+
+            tempvar range_check_ptr = range_check_ptr;
+        } else {
+            let is_length_gt_12 = is_le(12, length);
+            if (is_length_gt_12 != 0) {
+                // Case: mod_len > 12 * 1
+                // Copy limb1 with padding
+                let d1_bytes_len = length - 12;
+                memset_zero(bytes_ptr, d1_bytes_len - d1_actual_len);
+                memcpy(bytes_ptr + d1_bytes_len - d1_actual_len, d1_bytes, d1_actual_len);
+
+                // Copy limb0 with padding to exactly 12 bytes
+                memset_zero(bytes_ptr + d1_bytes_len, 12 - d0_actual_len);
+                memcpy(bytes_ptr + d1_bytes_len + 12 - d0_actual_len, d0_bytes, d0_actual_len);
+
+                tempvar range_check_ptr = range_check_ptr;
+            } else {
+                // Case: mod_len <= 12
+                // Copy only limb0 with padding
+                memset_zero(bytes_ptr, length - d0_actual_len);
+                memcpy(bytes_ptr + length - d0_actual_len, d0_bytes, d0_actual_len);
+
+                tempvar range_check_ptr = range_check_ptr;
+            }
+        }
+    }
+
+    tempvar result = Bytes(new BytesStruct(bytes_ptr, length));
+    return result;
+}
+
+func memset_zero{range_check_ptr}(dst: felt*, len: felt) {
```

Comment:
why do we need an extra util, wouldn't memset(dst, value=0, len=len) do the trick?

Reply:
absolutely right 
Will remove it